e2e/authentication/login.spec.js

```diff
@@ -0,0 +1,147 @@
+    });
+  });
+
+  test.describe('Rate limit on login', () => {
+    test('Should display a rate limit error message after 5 attempts to login', async ({
+      page,
+      browserName,
+    }) => {
+      await page.request.fetch('/api/config/ratelimit/enable', {
+        method: 'POST',
+        data: { value: true },
+      });
+
+      await page.getByLabel('Email*', { exact: true }).fill(`test@${browserName}.com`);
+      await page.getByLabel('Password*', { exact: true }).fill('wrongPassword');
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Too many requests, please try again later.')).toBeVisible();
+
+      await page.request.fetch('/api/config/ratelimit/enable', {
+        method: 'POST',
+        data: { value: false },
+      });
+    });
+  });
+
+  test.describe('Validation checks', () => {
+    test('A user should see a validation error when not passing in an email', async ({ page }) => {
+      await page.getByLabel('Password*', { exact: true }).fill('myTestPassw0rd');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Value is required')).toBeVisible();
+      await expect(await page.getByLabel('Email*', { exact: true })).toBeFocused();
+    });
+
+    test('A user should see a validation error when not passing in a password', async ({
+      page,
+    }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Value is required')).toBeVisible();
+      await expect(await page.getByLabel('Password*')).toBeFocused();
+    });
+
+    test('A user should see a validation error when entering a wrong email', async ({ page }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('wrongEmail@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('myTestPassw0rd');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Invalid credentials')).toBeVisible();
+    });
+
+    test('A user should see a validation error when entering a wrong password', async ({
+      page,
+    }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('wrongPassword');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Invalid credentials')).toBeVisible();
+    });
+  });
+
+  test.describe('Other actions than logging in', () => {
```

Comment:
Wondering if we could combine these too 🤔

Reply:
done [here](https://github.com/strapi/strapi/pull/15654/commits/165c30139e489e33749bf388f95477bfda6154c3)